docs/config/server-options.md

```diff
@@ -318,3 +318,20 @@ export default defineConfig({
   },
 })
 ```
+
+## server.sourcemapIgnoreList
+
+- **Type:** `(sourcePath: string, sourcemapPath: string) => boolean`
+- **Default:** `(sourcePath) => sourcePath.includes('/node_modules/')`
+
+Whether or not to ignore source files in the server sourcemap, used to populate the [`x_google_ignoreList` source map extension](https://developer.chrome.com/blog/devtools-better-angular-debugging/#the-x_google_ignorelist-source-map-extension).
+
+```js
+export default defineConfig({
+  server: {
+    // This is the default value, and will add all files with node_modules in their paths
+    // to the ignore list.
+    sourcemapIgnoreList: (sourcePath, sourcemapPath) => sourcePath.includes('/node_modules/')
```

Comment:
For https://github.com/rollup/rollup/pull/4877 I'm going with just `'node_modules'` (without the slashes) as otherwise it doesn't work on Windows. Not sure we always see normalized paths (with forward slashes) at this point, but it might be better (and more consistent) to just do `sourcePath.includes('node_modules')` here.